pkg/kritis/metadata/containeranalysis/containeranalysis.go

```diff
@@ -64,51 +64,73 @@ func (c Client) Close() {
 }
 
 //Vulnerabilities gets Package Vulnerabilities Occurrences for a specified image.
-func (c Client) Vulnerabilities(containerImage string, attestationAuthorities []kritisv1beta1.AttestationAuthority) ([]metadata.Vulnerability, error) {
+func (c Client) Vulnerabilities(containerImage string) ([]metadata.Vulnerability, error) {
+	occs, err := c.fetchVulnerabilityOccurrence(containerImage, PkgVulnerability)
+	if err != nil {
+		return nil, err
+	}
 	vulnz := []metadata.Vulnerability{}
-
-	for _, auth := range attestationAuthorities {
-		occs, err := c.fetchOccurrence(containerImage, PkgVulnerability, &auth)
-		if err != nil {
-			return nil, err
-		}
-		for _, occ := range occs {
-			if v := util.GetVulnerabilityFromOccurrence(occ); v != nil {
-				vulnz = append(vulnz, *v)
-			}
+	for _, occ := range occs {
+		if v := util.GetVulnerabilityFromOccurrence(occ); v != nil {
+			vulnz = append(vulnz, *v)
 		}
 	}
 
 	return vulnz, nil
 }
 
 //Attestations gets AttesationAuthority Occurrences for a specified image.
-func (c Client) Attestations(containerImage string, attestationAuthorities []kritisv1beta1.AttestationAuthority) ([]metadata.PGPAttestation, error) {
+func (c Client) Attestations(containerImage string, aa *kritisv1beta1.AttestationAuthority) ([]metadata.PGPAttestation, error) {
 	var p []metadata.PGPAttestation
 
-	for _, aa := range attestationAuthorities {
-		occs, err := c.fetchOccurrence(containerImage, AttestationAuthority, &aa)
+	occs, err := c.fetchAttestationOccurrence(containerImage, AttestationAuthority, aa)
+	if err != nil {
+		return nil, err
+	}
+	for _, occ := range occs {
+		pgp := util.GetPgpAttestationFromOccurrence(occ)
+		p = append(p, pgp)
+	}
+
+	return p, nil
+}
+
+func (c Client) fetchVulnerabilityOccurrence(containerImage string, kind string) ([]*grafeas.Occurrence, error) {
+	// Make sure container image valid and is a GCR image
+	if !isValidImageOnGCR(containerImage) {
+		return nil, fmt.Errorf("%s is not a valid image hosted in GCR", containerImage)
+	}
+
+	req := &grafeas.ListOccurrencesRequest{
+		Filter:   fmt.Sprintf("resource_url=%q AND kind=%q", util.GetResourceURL(containerImage), kind),
```

Comment:
Optional: Feel free to hard-code the kind.  There's the dedicated kind for vulnz after all.